pysigma/sigma_scan.py

```diff
@@ -40,20 +40,28 @@ def check_pair(event, key, value: 'Query') -> bool:
         return str(event[key]).lower() == value
 
 
-def find_matches(event: dict, search: 'DetectionField'):
+def find_matches(event: dict, search: 'DetectionField', match_all: bool = False):
     """
     Matches the items in the rule to the event. Iterates through the sections and if there's a list it iterates
     through that. Uses checkPair to see if the items in the list/dictionary match items in the event log.
 
     :param event: dict, event read from the Sysmon log
-    :param search: An object describin what sort of search to run
+    :param search: An object describing what sort of search to run
+    :param match_all: A bool indicating if we want all fields in list to hit
     :return: bool, whether or not we found a match
     """
     if search.list_search:
+        num_of_current_hits = 0
+        num_of_hits_required = len(search.list_search)
         for field in search.list_search:
             for event_key in event:
                 if check_pair(event, event_key, field):
```

Comment:
When match_all is set you can just return as soon as any check_pair fails instead of counting the number of successful check_pairs

Reply:
Not necessarily. There are many event_key value pairs in an event, and most of them contain irrelevant data, so returning False after any check_pair fails would be inaccurate. That being said, I've thought of a method to improve this just in case a field is something super common. Uno momento.

Reply:
The inner loop can also be changed to an any( ... ) wrapped list comprehension, which makes detecting a failure easier

Reply:
Sorry to clarify, not any check_pair failing, but anytime the inner loop fails to get any matches.